packages/react/src/hooks/useTheme.ts

```diff
@@ -1,10 +1,14 @@
-import { useContext } from 'react';
+import * as React from 'react';
 
-import { WebTheme } from '@aws-amplify/ui';
+import { defaultTheme, WebTheme } from '@aws-amplify/ui';
 import { AmplifyContext } from '../components/AmplifyProvider/AmplifyContext';
 
 export const useTheme = (): WebTheme => {
-  const { theme } = useContext(AmplifyContext);
+  const context = React.useContext(AmplifyContext);
 
-  return theme;
+  if (typeof context === 'undefined' || typeof context.theme === 'undefined') {
+    return defaultTheme;
+  }
+
+  return context.theme;
 };
```

Comment:
Since we are already changing some aspects of the hook, do you think it is better to use the existing `useAmplify` hook here instead of using `useContext` with the `AmplifyContext`? That was using the context would be at just one place. Once we do that, we can move the nullish check of context in that hook.

Reply:
agreed, will also be refactored to have a single source of truth